### PR TITLE
Agilis iv dc

### PR DESCRIPTION
**Pull request by dcmaverick**
Fixed the csv issue of not showing values. JIRA: https://jira.se.com/browse/FIRMWARE-24093

### CODE REVIEW

File: src/bacnet/basic/object/csv.c [hunk 1]
```diff
@@ -181,13 +181,13 @@ bool CharacterString_Value_Valid_Instance(uint32_t object_instance)
     unsigned index = 0; /* offset from instance lookup */
```
Comment (metelik) [inline src/bacnet/basic/object/csv.c:181]: it is legal in ansi-c to: 'unsigned index = CharacterString_Value_Instance_To_Index(object_instance);'

File: src/bacnet/basic/object/csv.c [hunk 2]
```diff
@@ -181,13 +181,13 @@ bool CharacterString_Value_Valid_Instance(uint32_t object_instance)
...
 
     index = CharacterString_Value_Instance_To_Index(object_instance);
-
-    if(index >= CSV_Max_Index) {
-        return false;
+    for (index = 0; index < CSV_Max_Index; index++) {
```
Comment (metelik) [inline src/bacnet/basic/object/csv.c:184]: the left hand side seems to be correct.

File: src/bacnet/basic/object/csv.c [hunk 3]
```diff
@@ -237,22 +233,22 @@ bool CharacterString_Value_Set(BACNET_OBJECT_LIST_INIT_T *pInit_data)
  * For a given object instance-number, read the present-value.
  *
  * @param  object_instance - object-instance number of the object
- * @param  present_value - Pointer to the new BACnet string value,
+ * @param  Name - Pointer to the new BACnet string value,
```
Comment (metelik) [inline src/bacnet/basic/object/csv.c:236]: left hand side...

File: src/bacnet/basic/object/csv.c [hunk 4]
```diff
@@ -477,20 +473,21 @@ bool CharacterString_Value_Object_Name(
...
+    size_t i = 0; /* loop counter */
     bool status = false; /* return value */
 
     index = CharacterString_Value_Instance_To_Index(object_instance);
     if (index < CSV_Max_Index) {
         status = true;
         /* FIXME: check to see if there is a matching name */
         if (new_name) {
-            strncpy(CSV_Descr[index].Name, new_name, sizeof(CSV_Descr[index].Name));
+            strcpy(CSV_Descr[index].Name, new_name);
```
Comment (alanj853) [inline src/bacnet/basic/object/csv.c:484]: Is strncpy not regarded as the safer one to use?

File: src/bacnet/basic/object/csv.c [hunk 5]
```diff
@@ -148,13 +148,13 @@ unsigned CharacterString_Value_Instance_To_Index(uint32_t object_instance)
 uint32_t CharacterString_Value_Index_To_Instance(unsigned index)
 {
 
-    if(index < CSV_Max_Index){
+    if(index < CSV_Max_Index) {
         return CSV_Descr[index].Instance;
     } else {
-        PRINT("index out of bounds %d", CSV_Descr[index].Instance);
+       PRINT("index out of bounds");
```
Comment (alanj853) [inline src/bacnet/basic/object/csv.c:154]: Would this not be useful debug to keep?

Comment (dcmaverick) [reply src/bacnet/basic/object/csv.c:154]: Same as above the other comment, I'm correcting them

File: src/bacnet/basic/object/csv.c (hunk 4 above)
Comment (metelik) [inline src/bacnet/basic/object/csv.c:484]: strncpy on the left hand side is better

**Review by metelik — CHANGES_REQUESTED** [review]
looks like the existent version is much better than the new one. the definite bug was around the apdu_len = .... line.... at read prop'

**Review by metelik — APPROVED** [review]
looks much better

**Review by alanj853 — APPROVED** [review]

File: src/bacnet/basic/object/csv.c [hunk 6]
```diff
@@ -405,7 +400,7 @@ static char *CharacterString_Value_Description(uint32_t object_instance)
         return CSV_Descr[index].Description;
     }
 
-    return NULL;
+    return 0;
```
[the remark this answers is not in the document]
Comment (adrigonzo) [reply src/bacnet/basic/object/csv.c:402]: why are we changing from NULL? Function returns a pointer, so seems correct.

**Pull request merged**